sources/settings-extensions/network/src/lib.rs

```diff
@@ -0,0 +1,98 @@
+/// The network settings will affect host service component's network behavior
+pub mod generate;
+
+use bottlerocket_settings_sdk::{GenerateResult, SettingsModel};
+use error::GenerateHostnameSnafu;
+use model_derive::model;
+use modeled_types::{EtcHostsEntries, SingleLineString, Url, ValidLinuxHostname};
+use snafu::ResultExt;
+
+#[model(impl_default = true)]
+struct NetworkSettingsV1 {
+    hostname: ValidLinuxHostname,
+    hosts: EtcHostsEntries,
+    https_proxy: Url,
+    // We allow some flexibility in NO_PROXY values because different services support different formats.
+    no_proxy: Vec<SingleLineString>,
+}
+
+type Result<T> = std::result::Result<T, error::Error>;
+
+impl SettingsModel for NetworkSettingsV1 {
+    type PartialKind = Self;
+    type ErrorKind = error::Error;
+
+    fn get_version() -> &'static str {
+        "v1"
+    }
+
+    fn set(_current_value: Option<Self>, _target: Self) -> Result<()> {
+        // Anything that parses as NetworkSettingsV1 is ok
+        Ok(())
+    }
+
+    fn generate(
+        existing_partial: Option<Self::PartialKind>,
+        _dependent_settings: Option<serde_json::Value>,
+    ) -> Result<GenerateResult<Self::PartialKind, Self>> {
+        let partial = existing_partial.unwrap_or_default();
+
+        // OpenQ: how to make this extension depend on netdog?
+        Ok(GenerateResult::Complete(NetworkSettingsV1 {
+            hostname: Some(
+                partial
+                    .hostname
+                    .unwrap_or(generate::generate_hostname().context(GenerateHostnameSnafu {})?),
+            ),
+            ..partial
+        }))
+    }
+
+    fn validate(_value: Self, _validated_settings: Option<serde_json::Value>) -> Result<()> {
+        Ok(())
+    }
+}
+
+mod error {
+    use super::generate;
+    use snafu::Snafu;
+
+    #[derive(Debug, Snafu)]
+    #[snafu(visibility(pub))]
+    pub enum Error {
+        #[snafu(display("Failed to generate hostname: {}", source))]
+        GenerateHostname {
+            source: generate::error::GenerateError,
+        },
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn test_serde_network() {
+        let test_json = r#"{
```

Comment:
Add a test where the JSON fails to parse due to an invalid hostname or some other invalid field. It helps document the way the type enforces correctness of its input.